File: src/backend/laravel-app/app/Models/Faute.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\Eleve;

/**
 * @OA\Schema(
 *     required={"libelle", "gravite","eleveId","regleId"},
 *     @OA\Xml(name="Faute"),
 *     @OA\Property(property="id", type="integer", readOnly=true, example="1"),
 *     @OA\Property(property="libelle", type="string", example="escalade"),
 *     @OA\Property(property="gravite", type="string", example="recidif"),
 *     @OA\Property(property="eleveId", type="integer", readOnly=true, example="1"),
 *     @OA\Property(property="regleId", type="integer", readOnly=true, example="1"),
 *     @OA\Property(property="eleve", type="object", ref="#/components/schemas/Eleve"),
 *     @OA\Property(property="regle", type="object", ref="#/components/schemas/Regle"),
 *     @OA\Property(property="created_at", ref="#/components/schemas/BaseModel/properties/created_at"),
 *     @OA\Property(property="updated_at", ref="#/components/schemas/BaseModel/properties/updated_at"),
 *     @OA\Property(property="deleted_at", ref="#/components/schemas/BaseModel/properties/deleted_at")
 * )
 *
 * @method static create(array $array)
 * @method static findOrFail($id)
 * @method static paginate(int $int)
 *
 * Class Regle
 *
 * @package App\Models
 */

class Faute extends Model
{
    use HasFactory;
    protected $fillable = [
        'id',
        'libelle',
        'gravite',
        'eleveId',
        'regleId',
        'eleve',
        'regle'
    ];
    public function regle()
    {
        return $this->belongsTo(Regle::class, 'regleId');
    }
    public function eleve()
    {
        return $this->belongsTo(Eleve::class, 'eleveId');
    }

    public function convocation()
    {
        return $this->belongsTo(Convocation::class);
    }
}

File: src/frontend/mobile/flutter_app/lib/common/models/faute.dart
import 'package:fltter_app/common/models/regle.dart';

class Faute {
  final int id;
  final String createdAt;
  final String updatedAt;
  final String ;
  final String gravite;
  final int regleId;
  final Regle regle;

  Faute({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.libelle,
    required this.gravite,
    required this.regleId,
    required this.regle,
  });

  factory Faute.fromJson(Map<String, dynamic> json) => Faute(
        id: json['id'],
        createdAt: json['created_at'],
        updatedAt: json['updated_at'],
        libelle: json['libelle'],
        gravite: json['gravite'],
        regleId: json['regleId'],
        regle: Regle.fromJson(json['regle']),
      );
}
